src/javascripts/components/tabs.js

```diff
@@ -1,133 +1,132 @@
-    if (!id) {
-      console.error('id is undefined')
-      return
-    }
-
-    var $example = $(id).parent()
-
-    // Reset state
-    $example.find('.js-tabs__item').removeClass('app-tabs__item--current')
-    $example.find('.js-tabs__heading').removeClass('app-tabs__heading--current')
-    $example.find('.js-tabs__item a').attr('aria-expanded', 'false')
-    $example.find('.js-tabs__container').hide().attr('aria-hidden', 'true')
-  },
-
-  // Attach aria attributes on load based on whether tabs have been set to open or closed
-  attachAriaAttributes: function (container) {
-    // Check if we have an id
-    if (!container) {
-      console.error('container is undefined')
-      return
-    }
-
-    var $example = $(container).parent()
-    var isTabOpen = $example.find('.js-tabs__item a').attr('aria-expanded')
+import 'govuk-frontend/vendor/polyfills/Function/prototype/bind'
+import 'govuk-frontend/vendor/polyfills/Element'
+import 'govuk-frontend/vendor/polyfills/Element/prototype/classList'
+import 'govuk-frontend/vendor/polyfills/Event'
+import common from 'govuk-frontend/common'
+
+var nodeListForEach = common.nodeListForEach
+
+var tabsItemClass = 'app-tabs__item'
+var tabsItemCurrentClass = tabsItemClass + '--current'
+var tabsItemJsClass = 'js-tabs__item'
+var headingItemClass = 'app-tabs__heading'
+var headingItemCurrentClass = headingItemClass + '--current'
+var headingItemJsClass = 'js-tabs__heading'
+var tabContainerHiddenClass = 'app-tabs__container--hidden'
+var tabContainerJsClass = '.js-tabs__container'
+var tabContainerNoTabsJsClass = 'js-tabs__container--no-tabs'
+var tabContainerWithCloseBtnClass = 'app-tabs__container--with-close-button'
+var allTabTogglers = '.' + tabsItemJsClass + ' a, ' + '.' + headingItemJsClass + ' a'
+var tabTogglersMarkedOpenClass = '.js-tabs__item--open a'
+var closeButtonClass = 'js-tabs__close'
+
+function AppTabs ($module) {
+  this.$module = $module
+  this.$allTabsContainers = this.$module.querySelectorAll(tabContainerJsClass)
+  this.$tabTriggers = this.$module.querySelectorAll(allTabTogglers)
+  this.$tabContainerCollection = this.$module.querySelectorAll(tabContainerJsClass)
+  this.$tabsTogglersMarkedOpen = this.$module.querySelectorAll(tabTogglersMarkedOpenClass)
+}
 
-    if (!isTabOpen) {
-      $example.find('.js-tabs__item a').attr('aria-expanded', 'false')
-      $example.find('.js-tabs__container').hide().attr('aria-hidden', 'true')
-    } else {
-      $example.find('.js-tabs__item a').attr('aria-expanded', 'true')
-      $example.find('.js-tabs__container').hide().attr('aria-hidden', 'false')
-    }
-  },
+AppTabs.prototype.init = function () {
+  if (!this.$module) {
+    return
+  }
+  // reset all tabs
+  this.resetTabs()
+  // add close buttons to each tab
+  this.addCloseBtn()
+  this.$module.addEventListener('click', this.handleClick.bind(this))
+
+  nodeListForEach(this.$tabsTogglersMarkedOpen, function ($tabToggler) {
+    $tabToggler.click()
```

Comment:
Non-blocking: I think we could execute the internal functions rather than relying on simulating user events.

Reply:
i opted to simulate a click, so that i could reuse `activateAndToggle` and `resetTabs` function without having to create a separate setup of these elements

Reply:
Yeah I think taking a look at how the GOV.UK frontend code does this would be useful, not a blocker tho'